### ISSUE TITLE
Sidebar does not update after name change

### ISSUE THREAD

**Issue by mhenninghaus77** [RocketChat/Rocket.Chat#22647]
This issue has been created with #22371 one month ago, but it has been closed without retesting and has not been opened after my comments.
Please either open #22371 or inform me to tell all the stuff in this ticket, too

**Comment by johncrisp** [comment]
Hi and thanks for reporting this.

We think this might be a slightly different issue.

Can you please test on the latest version fill out a bug template?

**Comment by mhenninghaus77** [comment]
no, for me it makes no difference for what reason and how a name has changed. only the display of the correct name in the sidebar is the desired result.
But I would be happy to fill in the data again here as well

**Comment by mhenninghaus77** [comment]
Description:

If a user name changes, why and however, then this changed name also appears correctly in the admin interface and for new chats. Old chats that are conducted with the user under his old user name are still displayed in the sidebar with the old name.

Steps to reproduce:

Chat with a User
Change user name
Chat again with the user, still the old username is displayed
Expected behavior:

Changes to the user name are adopted consistently everywhere

Actual behavior:

Username is correct in the admin interface and for new chats, for existing chats it remains the old name forever.

Server Setup Information:

Version of Rocket.Chat Server: 3.16
Operating System: Debian 10
Deployment Method: tar
Number of Running Instances: 1
DB Replicaset Oplog: yes
NodeJS Version: v12.22.1
MongoDB Version: 4.2.14
Client Setup Information

Desktop App or Browser Version: All Browsers and Apps
Operating System: -
Additional context

This problem is not new but has always existed.

**Comment by ankar84** [comment]
Confirm that bug on a 3.16.3 version.
Beside sidebar - chat header remain with old name.

**Comment by bbrauns** [comment]
Can confirm. A major issue when people get married and change their last name.

**Comment by ankar84** [comment]
> Can confirm. A major issue when people get married and change their last name.

Indeed! We have same problem. People complaining about that.

**Comment by jbisinotti** [comment]
can't reproduce this issue in the current server version

**Comment by mhenninghaus77** [comment]
but the problem is still there. freshly installed version 3.18.0, and still the wrong names are displayed.

But once again I am shown which functions 
with the 4.0 are deleted and are only included in the enterprise mode. 

it will be difficult for me to convince the responsible people in my company to switch to the enterprise variant when I see how some tickets are ignored here or the "customer" is degraded to a diligent version updater although he knows that nothing has changed in the problem.

**Comment by mhenninghaus77** [comment]
Any News on this?

**Comment by ankar84** [comment]
That script by @budakovda will fix that issue https://github.com/budakovda/custom_rocketchat/tree/main/rocketchat/fname
That issue related to https://github.com/RocketChat/Rocket.Chat/issues/23093
